src/host.cpp

```diff
@@ -924,17 +929,17 @@ namespace rhost {
         }
 
         void handle_binary_message(ws_connection_type::message_ptr msg) {
+            std::lock_guard<std::mutex> lock(binary_message_queue_mutex);
+
             if (binary_message_queue.size() != 1) {
-                fatal_error("Unexpected incoming client message - host was expecting a single pending binary message.");
+                fatal_error("Unexpected incoming client message - host was expecting a binary message.");
             }
 
             message incoming = binary_message_queue.front();
             const auto& raw = msg->get_raw_payload();
-            rhost::util::blob_slice slice(raw.data(), raw.data() + raw.size());
-            incoming.blob.push_back(std::move(slice));
+            incoming.blob.emplace_back(rhost::util::blob_slice(raw.data(), raw.data() + raw.size()));
```

Comment:
You don't need to invoke the constructor explicitly with `emplace` - it will automatically invoke it, and forward all the arguments. So you can just do:
```cpp
emplace_back(raw.data(), raw.data() + raw.size())
```